homeassistant/components/tts/media_source.py

```diff
@@ -64,9 +67,13 @@ def generate_media_source_id(
     if options is not None:
         params.update(options)
 
+    if all(isinstance(v, (str, int, float, bool)) for v in params.values()):
+        query = params
+    else:
+        query = {URL_QUERY_JSON: json.dumps(params)}
```

Comment:
We should specify the separators to avoid unnecessary whitespace:
```suggestion
        query = {URL_QUERY_JSON: json.dumps(params), separators=(',', ':')}
```